### ISSUE TITLE
Xcode 9 Export enterprise is not woking 

### ISSUE THREAD

**Issue by nageshkumarmishra** [fastlane/fastlane#11090]
bvkxtmyoftughvbzqbmjkwuqufjb/Build/Intermediates.noindex/ArchiveIntermediates/Sample/InstallationBuildProductsLocation/Applications/Sample.app[0m
[06:50:51]: [35m[33m▸[0m [39;1mTouching[0m Sample.app.dSYM[0m
[06:50:52]: [35m[33m▸[0m [39;1mArchive[0m Succeeded[0m
[06:50:52]: [33mGenerated plist file with the following values:[0m
[06:50:52]: ▸ [35m-----------------------------------------[0m
[06:50:52]: ▸ [35m{[0m
[06:50:52]: ▸ [35m  "method": "enterprise"[0m
[06:50:52]: ▸ [35m}[0m
[06:50:52]: ▸ [35m-----------------------------------------[0m
[06:50:52]: [4m[36m$ /usr/bin/xcrun /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.23.0/gym/lib/assets/wrap_xcodebuild/xcbuild-safe.sh -exportArchive -exportOptionsPlist '/var/folders/6g/08mp43t161xfmj452m2gtg980000gn/T/gym_config20171201-37009-l2rmb5.plist' -archivePath /Users/admin/Library/Developer/Xcode/Archives/2017-12-01/Sample\ 2017-12-01\ 06.47.23.xcarchive -exportPath '/var/folders/6g/08mp43t161xfmj452m2gtg980000gn/T/gym_output20171201-37009-19y5q8n' [0m[0m
RVM detected, forcing to use system ruby
Now using system ruby.
rbenv detected, removing env variables
rbenv: no such command `shell'
+ xcodebuild -exportArchive -exportOptionsPlist /var/folders/6g/08mp43t161xfmj452m2gtg980000gn/T/gym_config20171201-37009-l2rmb5.plist -archivePath '/Users/admin/Library/Developer/Xcode/Archives/2017-12-01/Sample 2017-12-01 06.47.23.xcarchive' -exportPath /var/folders/6g/08mp43t161xfmj452m2gtg980000gn/T/gym_output20171201-37009-19y5q8n
2017-12-01 06:50:53.332 xcodebuild[42785:2029596] [MT] IDEDistribution: -[IDEDistributionLogging _createLoggingBundleAtPath:]: Created bundle at path '/var/folders/6g/08mp43t161xfmj452m2gtg980000gn/T/Sample_2017-12-01_06-50-53.331.xcdistributionlogs'.
2017-12-01 06:50:54.662 xcodebuild[42785:2029596] [MT] IDEDistribution: Step failed: <IDEDistributionSigningAssetsStep: 0x7fd38a489290>: Error Domain=IDEDistributionSigningAssetStepErrorDomain Code=0 "Locating signing assets failed." UserInfo={NSLocalizedDescription=Locating signing assets failed., IDEDistributionSigningAssetStepUnderlyingErrors=(
"Error Domain=IDEProfileLocatorErrorDomain Code=1 \"No profiles for 'com.sample.sample' were found\" UserInfo={NSLocalizedDescription=No profiles for 'com.sample.sample' were found, NSLocalizedRecoverySuggestion=Xcode couldn't find any iOS In House provisioning profiles matching 'com.sample.tae2f92e0'. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild.}"
)}
error: exportArchive: No profiles for 'com.sample.sample' were found

Error Domain=IDEProfileLocatorErrorDomain Code=1 "No profiles for 'com.sample.sample' were found" UserInfo={NSLocalizedDescription=No profiles for 'com.sample.sample' were found, NSLocalizedRecoverySuggestion=Xcode couldn't find any iOS In House provisioning profiles matching 'com.sample.sample'. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild.}

** EXPORT FAILED **
[06:50:54]: [31mExit status: 70[0m
[06:50:54]: [33m📋  For a more detailed error log, check the full log at:[0m
[06:50:54]: [33m📋  /Users/admin/Library/Logs/gym/Sample-Sample.log[0m
+------------------+----------+
|        [33mLane Context[0m         |
+------------------+----------+
| DEFAULT_PLATFORM | ios      |
| VERSION_NUMBER   | 1.2.1    |
| BUILD_NUMBER     | 1.2.1    |
| PLATFORM_NAME    | ios      |
| LANE_NAME        | ios beta |
+------------------+----------+
[06:50:54]: [31mError packaging up the application[0m

+------+-------------------------------------+-------------+
|                     [32mfastlane summary[0m                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying required fastlane version | 0           |
| 2    | default_platform                    | 0           |
| 3    | increment_version_number            | 0           |
| 4    | get_version_number                  | 0           |
| 5    | increment_build_number              | 0           |
| 6    | cocoapods                           | 4           |
| 💥   | [31mgym[0m                                 | 214         |
+------+-------------------------------------+-------------+

[06:50:54]: [31mfastlane finished with errors[0m
[31m
[!] Error packaging up the application[0m
Build step 'Execute managed script' marked build as failure
Performing Post build task...
Match found for : : True
Logical operation result is TRUE

----
Above is the error message coming when I am using gym(scheme: "Sample",export_method: "enterprise")

-- Please let me know how can I fix this issue using Xcode 9.

**Comment by janpio** [comment]
> Error Domain=IDEProfileLocatorErrorDomain Code=1 "No profiles for 'com.sample.sample' were found" UserInfo={NSLocalizedDescription=No profiles for 'com.sample.sample' were found, NSLocalizedRecoverySuggestion=Xcode couldn't find any iOS In House provisioning profiles matching 'com.sample.sample'. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild.}

This looks wrong.

**Comment by PetrosIchor** [comment]
Ran into this myself recently.  Xcode 9 changes some ways you do signing, for adhoc/appstore you can simply provide an export options.plist with the configuration you want.  However I haven't been successful doing this for Enterprise. 

Instead, I have to modify the xcodeproj to set
Manual
Teamid
Provisioning profile specifier
Code signing identity

Personally, I used the update_xcode_project plugin. 

Setting those params as a step in fastlane fixes my export issues.

Sorry my response is short, on mobile

**Comment by oronbz** [comment]
Same as https://github.com/fastlane/fastlane/issues/11059

**Comment by oronbz** [comment]
@PetrosIchor this is not related to Xcode 9, this is a regression bug. It worked perfectly on 2.66.

**Comment by gcmleon** [comment]
I also had this issue trying to use an Ad-Hoc profile with the latest Fastlane version, 2.68.2.
Thanks to @oronbz comment, I changed to 2.66.0 and it worked.

**Comment by oronbz** [comment]
@gcmleon , this issue was resolved for me on 2.68.2, make sure you specify `export_method: "ad-hoc"` on `gym`.

**Comment by ohayon** [comment]
If it's cool with you folks, I am going to close this issue. Let us know if there are still issues and we can definitely reopen it!

**Comment by gcmleon** [comment]
@oronbz @ohayon Unfortunately that did not work for me. I tried specifying the method (also "ad-hoc" in my case) and specifying the provisioning profile name in the export_options.
Furthermore, I experimented changing the signing style and using the flag -allowProvisioningUpdates based on the discussion in #11073 but still didn't work. However, like I said, using 2.66.0 with my original code worked well.
